www/guides/react.md

```diff
@@ -0,0 +1,371 @@
+
+Now that you have a basic project up and running, it's time to install React. To install React, run the following command in your project directory:
+
+```bash
+npm install react react-dom --save
+```
+
+> 💡 Tip: in this guide commands use `--save` for frontend libraries and `--save-dev` for your developer tools to keep them organized separately. This is totally up to you, though.
+
+## Create your first React component
+
+React relies on a special templating language called JSX. Snowpack has built in support for JSX in files with the `.jsx` extension. That means that there's no plugins or configuration needed to write your first React component. Rename `index.js` file to `index.jsx` so that Snowpack knows it's now a JSX file:
+
+```bash
+mv index.js index.jsx
+```
+
+> 💡 Tip: you do not need to update your `index.html` script tag to point to `index.jsx`. Because JSX isn't supported in the browser, Snowpack builds your file to a runnable `.js` file. This is good to remember when you're referencing built files in HTML `<script src="">` and `<link href="">` elements.
+
+You can now import React in `index.jsx` and add a simple test component just to make sure it's working:
+
+```diff
+/* Add JavaScript code here! */
+- console.log('Hello World! You did! Welcome to Snowpack :D');
++import React from 'react';
++import ReactDOM from 'react-dom';
++ReactDOM.render(<div>"HELLO REACT"</div>, document.getElementById('root'));
+```
+
+Since the React code is rendering into an element with the ID `root`, you'll need to add that to `index.html`:
+
+```diff
+  <body>
+-    <h1>Welcome to Snowpack!</h1>
++    <div id="root"></div>
+    <script type="module" src="/index.js"></script>
+```
+
+Image: screenshot of the project, which shows "HELLO REACT" on a white background
+
+You've just created your first React component in Snowpack!
+
+## Customize your project layout
+
+Since you'll be adding a bunch of new files, you probably don't want them crowding up your top-level root directly. Snowpack is flexible enough to support whatever project layout that you prefer. In this guide, you'll learn how to use a popular project pattern from the React community.
+
+In this pattern the `src` directory is for components and their associated files. Then a `public` directory is for static assets like images, fonts, icons, and sometimes CSS.
+
+```bash
+📁 src
+↳ index.jsx
+📁 public
+↳ index.css
+↳ index.html
+```
+
+Use your favorite visual editor to rearrange and rename, or run these commands in the terminal:
+
+```bash
+mkdir src
+mkdir public
+mv index.jsx src/index.jsx
+mv index.html public/index.html
+mv index.css public/index.css
+```
+
+Once your new project structure is in place, you need to tell Snowpack about it. Every Snowpack project comes with a `snowpack.config.js` file for any configuration that you might need. Right now, you should see a mostly empty, bare-bones configuration file. Add this to the empty `mount` object:
```

Comment:
I like this explainer section, but I think as a beginner I’d be confused why `src` and why `public`. What is the purpose of each?

Also by default, Snowpack scans the entire folder; `mount` just limits the scope. So it may be more accurate now to say “we may have files we don’t want to publish online, so let’s tell Snowpack to only build specific folders.” You may also want to mention this can be a speed boost, if we’re only scanning part of the project.